src/app/clusters/mode-base-server/mode-base-server.h

```diff
@@ -0,0 +1,198 @@
+    void UpdateCurrentMode(uint8_t aNewMode);
+
+    // Attribute getters
+    DataModel::Nullable<uint8_t> GetStartUpMode() const;
+    DataModel::Nullable<uint8_t> GetOnMode() const;
+    uint8_t GetCurrentMode() const;
+    EndpointId GetEndpointId() const { return mEndpointId; }
+
+    /**
+     * Returns true if the feature is supported.
+     * @param feature the feature to check.
+     */
+    bool HasFeature(Feature feature) const;
+
+    /**
+     * This function returns true if the mode value given matches one of the supported modes, otherwise it returns false.
+     * @param mode
+     */
+    bool IsSupportedMode(uint8_t mode);
+
+private:
+    EndpointId mEndpointId{};
+    ClusterId mClusterId{};
+
+    // Attribute data store
+    uint8_t mCurrentMode;
+    DataModel::Nullable<uint8_t> mStartUpMode;
+    DataModel::Nullable<uint8_t> mOnMode;
+
+    uint32_t mFeature;
+
+    /**
+     * This checks to see if this clusters instance is a valid ModeBase aliased cluster based on the AliasedClusters list.
+     * @return true if the clusterId of this instance is a valid ModeBase cluster.
+     */
+    bool isAliasCluster() const;
+
+    /**
+     * Internal change-to-mode command handler function.
+     */
+    void handleChangeToMode(HandlerContext & ctx, const Commands::ChangeToMode::DecodableType & req);
+
+public:
+    /**
+     * Creates a mode base cluster instance. The Init() function needs to be called for this instance to be registered and
+     * called by the interaction model at the appropriate times.
+     * @param aEndpointId The endpoint on which this cluster exists. This must match the zap configuration.
+     * @param aClusterId The ID of the ModeBase aliased cluster to be instantiated.
+     * @param aFeature The bitmask value that identifies which features are supported by this instance.
+     */
+    Instance(EndpointId aEndpointId, ClusterId aClusterId, uint32_t aFeature) :
+        CommandHandlerInterface(Optional<EndpointId>(aEndpointId), aClusterId),
+        AttributeAccessInterface(Optional<EndpointId>(aEndpointId), aClusterId)
+    {
+        mEndpointId = aEndpointId;
+        mClusterId  = aClusterId;
+        mFeature    = aFeature;
+    }
+
+    ~Instance() override { ModeBaseAliasesInstanceMap.erase(mClusterId); }
+
+    template <typename RequestT, typename FuncT>
+    void HandleCommand(HandlerContext & handlerContext, FuncT func);
+
+    // The following functions should be overridden by the SDK user to implement the business logic of their application.
+
+    /**
+     * This init function will be called during the ModeBase server initialization after the Instance information has been
+     * validated and the Instance has been registered. This can be used to initialise app logic.
+     */
+    virtual CHIP_ERROR AppInit() = 0;
+
+    /**
+     * Returns the number of modes managed by this instance.
+     */
+    virtual uint8_t NumberOfModes();
```

Comment:
Should this not be `= 0` ? Same for all the other mandatory ones.

Reply:
Done [here](https://github.com/project-chip/connectedhomeip/pull/27504/commits/be64a611f136960530ada54407c47964c0909176).